### PR TITLE
Windows Server 2008 R2 TerminalServices BEX Access Violation

### PR DESCRIPTION
**Pull request by JJO30**
Access Violation of BEX or APPCRASH occurrs on Windows Server 2008 R2 with Terminal Service

Hi,
I have AppCrash with CefSharpSubProcess in Windows Server 2008 R2 with Terminal Service.
I can not disable Data Execution Prevention (DEP) for CefSharpSubProcess.exe.
I have this message : " the execution of data protection must be activated to execute this program. You can not disable the protection of data execution for this program."

I think to activate TSAWARE for CefSharpSubprocess.exe. See : https://support.microsoft.com/en-us/kb/2279689 
Option 3

Thanks

### CODE REVIEW

File: CefSharp.BrowserSubprocess/CefSharp.BrowserSubprocess.csproj [hunk 1]
```diff
@@ -120,4 +122,4 @@
   <Target Name="AfterBuild">
   </Target>
   -->
-</Project>
\ No newline at end of file
+</Project>
```
Comment (amaitland) [inline CefSharp.BrowserSubprocess/CefSharp.BrowserSubprocess.csproj]: Ideally revert unrelated changes, :+1:

File: CefSharp.BrowserSubprocess/CefSharp.BrowserSubprocess.csproj [hunk 2]
```diff
@@ -111,7 +111,9 @@
     <PostBuildEvent>if $(PlatformName) == x86 (
   call "$(DevEnvDir)..\tools\vsvars32.bat"
   editbin /largeaddressaware $(TargetPath)
-)</PostBuildEvent>
+)
+call "$(DevEnvDir)..\tools\vsvars32.bat"
```
Comment (amaitland) [inline CefSharp.BrowserSubprocess/CefSharp.BrowserSubprocess.csproj]: I think it's probably safer to call `vsvars64.bat` when building the `x64` version.

So something like below?

```
if $(PlatformName) == x86 (
  call "$(DevEnvDir)..\tools\vsvars32.bat"
  editbin /largeaddressaware /TSAWARE $(TargetPath)
) else (
  call "$(DevEnvDir)..\tools\vsvars64.bat"
  editbin /TSAWARE $(TargetPath)
)
```

**Comment by amaitland** [comment]
@JJO30 Great, thanks :+1: Comments inline :smile:

File: CefSharp.BrowserSubprocess/CefSharp.BrowserSubprocess.csproj (hunk 2 above)
Comment (JJO30) [inline CefSharp.BrowserSubprocess/CefSharp.BrowserSubprocess.csproj]: Yes I agree. 
I have to edit the code or you did it ?
Thanks

Comment (amaitland) [inline CefSharp.BrowserSubprocess/CefSharp.BrowserSubprocess.csproj]: You can push subsequent commits with the proposed changes.

File: CefSharp.BrowserSubprocess/CefSharp.BrowserSubprocess.csproj [hunk 3]
```diff
@@ -110,8 +110,11 @@
   <PropertyGroup>
     <PostBuildEvent>if $(PlatformName) == x86 (
   call "$(DevEnvDir)..\tools\vsvars32.bat"
-  editbin /largeaddressaware $(TargetPath)
-)</PostBuildEvent>
+  editbin /largeaddressaware $(TargetPath)  /TSAWARE $(TargetPath)
```
Comment (amaitland) [inline CefSharp.BrowserSubprocess/CefSharp.BrowserSubprocess.csproj]: do you need to specify `TargetPath` for each param? That seems like an unexpected behavior (I haven't looked at the doc).

**Comment by amaitland** [comment]
@JJO30 When your done making changes, best if you can post a comment here as `GitHub` doesn't provide any sort of notification as you pushed more commits.

File: CefSharp.BrowserSubprocess/CefSharp.BrowserSubprocess.csproj (hunk 3 above)
Comment (JJO30) [inline CefSharp.BrowserSubprocess/CefSharp.BrowserSubprocess.csproj]: I think it is not necessary
https://msdn.microsoft.com/en-us/library/8e77sz74.aspx

**Comment by amaitland** [comment]
@JJO30 Great, thanks. Soon as I've had a chance to build locally I'll merge your changes (I'll revert the line break change at the end, ideally in future you'd exclude those from your commits please). Will keep you posted.

**Comment by amaitland** [comment]
@JJO30 I'm guessing you didn't test the last set of changes you made? The `x64` builds weren't working. This has been merged with the following commit to make it work https://github.com/cefsharp/CefSharp/commit/bccfc17e3ed132fb9cd77abc3434dfb5a8093be9

**Comment by JJO30** [comment]
I had not tested build in x64, only x32, sorry. 
I discovered visual studio. But, you even changed the previous instruction using " vsvar32 " in your code ?
I'll see for 64bits instruction.

**Comment by amaitland** [comment]
> I discovered visual studio. But, you even changed the previous instruction using " vsvar32 " in your code ?

Any your questions is?

**Comment by JJO30** [comment]
Sorry I only just saw your comment in your merge : "Simplify and call editbin directly instead"
